src/code/z_lifemeter.c

```diff
@@ -1,17 +1,434 @@
+    s32 pad2;
+    f32 lifesize = interfaceCtx->lifeSizeChange * 0.1f;
+    u32 curCombineModeSet = 0;
+    TexturePtr temp = NULL;
+    s32 ddCount = gSaveContext.inventory.dungeonKeys[9] - 1;
+
+    OPEN_DISPS(gfxCtx);
+
+    if ((gSaveContext.health % 0x10) == 0){
+        fullHeartCount--;
+    }
+    offsetY = 0.0f;
+    offsetX = 0.0f;
+    curColorSet = -1;
+
+    for (i = 0; i < healthCapacity; i++) {
+        if ((ddCount < 0) || (ddCount < i)) {
+            if (i < fullHeartCount) {
+                if (curColorSet != 0) {
+                    curColorSet = 0;
+                    gDPPipeSync(OVERLAY_DISP++);
+                    gDPSetPrimColor(OVERLAY_DISP++, 0, 0, interfaceCtx->heartsPrimR[0], interfaceCtx->heartsPrimG[0],
+                                    interfaceCtx->heartsPrimB[0], interfaceCtx->healthAlpha);
+                    gDPSetEnvColor(OVERLAY_DISP++, interfaceCtx->heartsEnvR[0], interfaceCtx->heartsEnvG[0],
+                                   interfaceCtx->heartsEnvB[0], 255);
+                }
+            } else if (fullHeartCount == i) {
+                if (curColorSet != 1) {
+                    curColorSet = 1;
+                    gDPPipeSync(OVERLAY_DISP++);
+                    gDPSetPrimColor(OVERLAY_DISP++, 0, 0, interfaceCtx->beatingHeartPrim[0],
+                                    interfaceCtx->beatingHeartPrim[1], interfaceCtx->beatingHeartPrim[2],
+                                    interfaceCtx->healthAlpha);
+                    gDPSetEnvColor(OVERLAY_DISP++, interfaceCtx->beatingHeartEnv[0], interfaceCtx->beatingHeartEnv[1],
+                                   interfaceCtx->beatingHeartEnv[2], 255);
+                }
+            } else if (fullHeartCount < i) {
+                if (curColorSet != 2) {
+                    curColorSet = 2;
+                    gDPPipeSync(OVERLAY_DISP++);
+                    gDPSetPrimColor(OVERLAY_DISP++, 0, 0, interfaceCtx->heartsPrimR[0], interfaceCtx->heartsPrimG[0],
+                                    interfaceCtx->heartsPrimB[0], interfaceCtx->healthAlpha);
+                    gDPSetEnvColor(OVERLAY_DISP++, interfaceCtx->heartsEnvR[0], interfaceCtx->heartsEnvG[0],
+                                   interfaceCtx->heartsEnvB[0], 255);
+                }
+            } else {
+                if (curColorSet != 3) {
+                    curColorSet = 3;
+                    gDPPipeSync(OVERLAY_DISP++);
+                    gDPSetPrimColor(OVERLAY_DISP++, 0, 0, interfaceCtx->heartsPrimR[1], interfaceCtx->heartsPrimG[1],
+                                    interfaceCtx->heartsPrimB[1], interfaceCtx->healthAlpha);
+                    gDPSetEnvColor(OVERLAY_DISP++, interfaceCtx->heartsEnvR[1], interfaceCtx->heartsEnvG[1],
+                                   interfaceCtx->heartsEnvB[1], 255);
+                }
+            }
+
+            if (i < fullHeartCount) {
+                heartTex = &D_02000400;
+            } else if (fullHeartCount == i) {
```

Comment:
```suggestion
            } else if (i == fullHeartCount) {
```
Would this still  match?